test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ValidateTests.cs

```diff
@@ -187,11 +191,133 @@ public async void GivenAValidateResourceToUSCoreProfile_WhenTheResourceProfileIs
+                {
+                    Assert.True(paramComponenet.Value.ToString() == "true");
+                }
+
+                break;
+            }
+        }
+
+        [SkippableTheory]
+        [InlineData("ValueSet/birthsex/$validate-code", "http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender", "F", "IncorrectDisplay")]
+        [InlineData("ValueSet/birthsex/$validate-code", "http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender", "IncorrectCode", "")]
+        [InlineData("ValueSet/us-core-narrative-status/$validate-code", "http://hl7.org/fhir/narrative-status", "generated", "generate")] // Display should be "Generated"
+        [InlineData("ValueSet/us-core-narrative-status/$validate-code", "http://hl7.org/fhir/narrative-status", "Addition")] // Code should be "additional"
+        [InlineData("CodeSystem/example/$validate-code", "http://hl7.org/fhir/CodeSystem/example", "chol-mmol", "SChol")] // Display should "SChol (mmol/L)" May fail as CodeSystem resource changed STU3 -> R4 and the TS is on R4
+        [InlineData("CodeSystem/example/$validate-code", "http://hl7.org/fhir/CodeSystem/example", "acme", "")] // Code should be "acme-plasma" May fail as CodeSystem resource changed STU3 -> R4 and the TS is on R4
+        public async void GivenInValidCodeorDisplayFromKnownValueSet_ThenFalseParameterIsReturned(string path, string system, string code, string display = null)
+        {
+            Skip.If(string.IsNullOrEmpty(_fhirServer.Configuration["FhirServer:Operations:Validate:ExternalTerminologyServer"]));
+            Parameters resultParam = await _client.ValidateCodeGETdAsync(path, system, code, display);
+            foreach (var paramComponenet in resultParam)
+            {
+                if (paramComponenet.Key == "result")
+                {
+                    Assert.True(paramComponenet.Value.ToString() == "false");
+                }
+
+                break;
+            }
+        }
+
+        [SkippableTheory]
+        [InlineData("ValueSet/birthsex/$validate-code", "http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender", "", "IncorrectDisplay")]
+        [InlineData("ValueSet/us-core-narrative-status/$validate-code", "", "generated", "generated")]
+        [InlineData("CodeSystem/example/$validate-code", "", "chol-mmol", "SChol (mmol/L)")]
+        [InlineData("CodeSystem/example/$validate-code", "http://hl7.org/fhir/CodeSystem/example", "", "")]
+        public async void GivenInvalidRequestParams_ValidateCodeThrowsExcept(string path, string system, string code, string display = null)
+        {
+            Skip.If(string.IsNullOrEmpty(_fhirServer.Configuration["FhirServer:Operations:Validate:ExternalTerminologyServer"]));
+            await Assert.ThrowsAsync<FhirException>(async () => await _client.ValidateCodeGETdAsync(path, system, code, display));
+        }
+
+        [SkippableFact]
+        public async void GivenNoExternalTerminologyEndpoint_ThrowBadRequestException()
```

Comment:
I don't like this test. It only runs if we are testing against a server that doesn't support the terminology operations, which we won't be doing once this is merged to main. I don't think this test is needed.

Reply:
Removed test.